ios/FluentUI/Drawer/DrawerController.swift

```diff
@@ -801,14 +812,24 @@ open class DrawerController: UIViewController {
             } else if offset <= -Constants.resizingThreshold {
                 if isExpanded {
                     if originalDrawerHeight + offset <= normalDrawerHeight - Constants.resizingThreshold {
-                        presentingViewController?.dismiss(animated: true)
+                        if delegate?.shouldDismissDrawer?(self) == false {
```

Comment:
could we cache this value to local variable. this if else is getting way to complicated. is there a way to simplify? reorder, break it to functions/blocks ...

Reply:
I have made a suggestion below that will hopefully help on this topic.

Reply:
created `dismissPresentingViewController` keeping the code clean.